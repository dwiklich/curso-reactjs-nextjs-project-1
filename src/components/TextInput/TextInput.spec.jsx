import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { TextInput } from ".";

// const props = {
//   searchValue: 2,
//   handleChange: 1,
// };

describe("<TextInput />", () => {
  test("should have a value of searchValue", () => {
    const fn = jest.fn();

    render(
      <TextInput handleChange={fn} searchValue={"texto digitado do input"} />
    );

    const input = screen.getByPlaceholderText(/Type your search/i);

    expect(input).toHaveProperty("value", "texto digitado do input");
    expect(input.value).toBe("texto digitado do input");
  });

  test("should call handleChange function on each key pressed", () => {
    const fn = jest.fn();

    render(<TextInput handleChange={fn} />);

    const input = screen.getByPlaceholderText(/Type your search/i);

    const value = "o valor";

    userEvent.type(input, value);

    expect(input).toHaveAttribute("id", "pesquisa");
    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  test("should match snapshot", () => {
    const fn = jest.fn();

    const { container } = render(
      <TextInput handleChange={fn} searchValue={"texto digitado do input"} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
