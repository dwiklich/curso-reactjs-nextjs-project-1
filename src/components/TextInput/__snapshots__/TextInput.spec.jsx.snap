// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  class="text-input"
  id="pesquisa"
  name="procura"
  placeholder="Type your search"
  type="search"
  value="texto digitado do input"
/>
`;
